TombEngine/Game/Lara/lara_flare.cpp

```diff
@@ -318,6 +318,8 @@ void CreateFlare(ItemInfo& laraItem, GAME_OBJECT_ID objectID, bool isThrown)
 	flareItem.RoomNumber = laraItem.RoomNumber;
 
 	auto pos = GetJointPosition(&laraItem, LM_LHAND, Vector3i(-16, 32, 42));
+	if (objectID == ID_BURNING_TORCH_ITEM && laraItem.Animation.ActiveState == LS_CROUCH_IDLE)
```

Comment:
Which fix is this block for?

Reply:
When Lara crawls and tried to throw the torch, before the torch was colliding with the floor and bouncing backwards, that's because when the object was spawned, it did it slightly under the floor.

This block adds an offseet towards up so the torch is not created under the floor anymore, so it moves toward forward.